### PR TITLE
100% on ledger_apis module

### PR DESCRIPTION
**Pull request by Totoual**
## Fixes

100% coverage on aea/cryptos/ledger_api module.

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_crypto/test_ledger_apis.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from unittest import mock
 
 import pytest
-from hexbytes import HexBytes
+from hexbytes import HexBytes   # type: ignore
```
Comment (marcofavorito) [inline tests/test_crypto/test_ledger_apis.py]: It is OK but would be better to add an item in `setup.cfp`:

```
[mypy-hexbytes]
ignore_missing_imports = True
```

File: tests/test_crypto/test_ledger_apis.py [hunk 2]
```diff
@@ -48,56 +54,102 @@ def test_initialisation(self):
...
+        balance = ledger_apis.token_balance(FETCHAI, eth_address)
+        assert balance == 0, "This must be 0 since the address is wrong"
+        balance = ledger_apis.token_balance(ETHEREUM, fet_address)
+        assert balance == 0, "This must be 0 since the address is wrong"
+
+        with pytest.raises(AssertionError):
+            balance = ledger_apis.token_balance("UNKNOWN", fet_address)
+            assert balance == 0, "Unknown identifier so it will return 0"
+
+    def test_transfer(self):
```
Comment (marcofavorito) [inline tests/test_crypto/test_ledger_apis.py]: could this test be split in `test_transfer_fetchai` and `test_transfer_ethereum`

File: tests/test_crypto/test_ledger_apis.py [hunk 3]
```diff
@@ -48,56 +54,102 @@ def test_initialisation(self):
...
+                    ledger_apis.apis[ETHEREUM].eth.sendRawTransaction(signed.rawTransaction)
+                result = HexBytes('0xf85f808082c35094d898d5e829717c72e7438bad593076686d7d164a80801ba005c2e99ecee98a12fbf28ab9577423f42e9e88f2291b3acc8228de743884c874a077d6bc77a47ad41ec85c96aac2ad27f05a039c4787fca8a1e5ee2d8c7ec1bb6a')
+                with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth, 'sendRawTransaction',
+                                       return_value=result):
+                    with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth, "getTransactionReceipt",
+                                           return_value=b'0xa13f2f926233bc4638a20deeb8aaa7e8d6a96e487392fa55823f925220f6efed'):
+                        tx_digest = ledger_apis.transfer(ETHEREUM, eth_obj, eth_address, amount=10, tx_fee=200000)
+                        assert tx_digest is not None
+
+    def test_is_tx_settled(self):
```
Comment (marcofavorito) [inline tests/test_crypto/test_ledger_apis.py]: the same as above, would be better to split this test in multiple tests. At least the ones regarding `fetchai` and the others regarding `ethereum`.

File: aea/crypto/ledger_apis.py [hunk 4]
```diff
@@ -162,12 +162,12 @@ def transfer(self, identifier: str, crypto_object: Crypto, destination_address:
                     logger.info("transaction validated - exiting")
```
Comment (marcofavorito) [inline aea/crypto/ledger_apis.py]: regarding the `print` statements above:
- `print("TX Hash: ", hex_value.hex())` should be `logger.info("TX Hash {}".format(hex_value.hex()))`
- ` print("connect_to ...)` should be removed. It is an interesting feature to provide the link to block explorers, but we're not there yet.

File: aea/crypto/ledger_apis.py [hunk 5]
```diff
@@ -112,7 +112,7 @@ def token_balance(self, identifier: str, address: str) -> int:
             except Exception:
                 logger.warning("An error occurred while attempting to get the current balance.")
                 balance = 0
-        else:           # pragma: no cover
+        else:       # pragma: no cover
             balance = 0
```
Comment (marcofavorito) [inline aea/crypto/ledger_apis.py]: Question: in the case when the Ledger id is not known, shouldn't this raise an exception instead? Or return `None` (in this case, change the return value to `Optional[int]` in the signature).

Comment (DavidMinarsch) [reply aea/crypto/ledger_apis.py]: +1 for exception!

File: aea/crypto/ledger_apis.py [hunk 6]
```diff
@@ -162,12 +162,12 @@ def transfer(self, identifier: str, crypto_object: Crypto, destination_address:
...
                     break
-                except web3.exceptions.TransactionNotFound:
+                except web3.exceptions.TransactionNotFound:     # pragma: no cover
                     logger.info("transaction not found - sleeping for 3.0 seconds")
                     time.sleep(3.0)
 
             return tx_digest
-        else:                   # pragma: no cover
+        else:           # pragma: no cover
             tx_digest = None
```
Comment (marcofavorito) [inline aea/crypto/ledger_apis.py]: What about this? Should there be an exception too?

**Pull request merged**